File: src/Functional/Elixir/patterns/lazy_initialization.exs
get = fn
  nil -> {7, 1}
  value -> {value, 0}
end

{value, first_initializations} = get.(nil)
{_value, second_initializations} = get.(value)

unless first_initializations + second_initializations == 1 do
  raise "Lazy"
end

File: src/Functional/Erlang/patterns/lazy_initialization.erl
-module(lazy_initialization).
-export([main/0]).

initialize(undefined) -> {7, 1};
initialize(Value) -> {Value, 0}.

main() ->
    {Value, FirstInitializations} = initialize(undefined),
    {7, SecondInitializations} = initialize(Value),
    1 = FirstInitializations + SecondInitializations,
    ok.
